### PR TITLE
Sobhi week2

### PR DESCRIPTION
**Pull request by Sob7i**

### CODE REVIEW

File: Week2/todo_app/program.js [hunk 1]
```diff
@@ -62,11 +123,121 @@ dbConnection.connect(function(err) {
...
+
+dbConnection.end()
+});
+
+
+
+
+
+
+// Week 3 Homeworks
```
Comment (njaradat) [inline Week2/todo_app/program.js]: Hi :) Can you add the ER diagram for this too?

File: Week2/todo_app/program.js [hunk 2]
```diff
@@ -21,39 +21,100 @@ class TodoModel {
         });
     }
 
-    create(description, callback) {
+    create(description, userId, callback) {
         // Write code and query to create a new TODO item
+        const createTodoItem = `INSERT INTO todo_items(text, is_completed, user_id) 
+                                VALUES (${dbConnection.escape(description)}, "0", ${dbConnection.escape(userId)})`;
```
Comment (zhaobn) [inline Week2/todo_app/program.js]: We talked about using prepared statement provided my mysql.js to make you write less code in escaping, do check it out :)

Comment (zhaobn) [inline Week2/todo_app/program.js]: Does string type work in `is_completed` column?

File: Week2/todo_app/program.js [hunk 3]
```diff
@@ -21,39 +21,100 @@ class TodoModel {
...
+        }
+        callback(null, results);
+    });
+
+}
+
+markCompleted(todoItemId, callback) {
+    // Write code to mark a TODO item as completed
+    const completeTODOItem = `UPDATE todo_items SET is_completed = TRUE 
+    WHERE ${dbConnection.escape(todoItemId)} = id`;
```
Comment (zhaobn) [inline Week2/todo_app/program.js]: This is not correct sql syntax. Please check out how WHERE clauses work.

File: Week2/todo_app/program.js [hunk 4]
```diff
@@ -62,11 +123,121 @@ dbConnection.connect(function(err) {
...
+        if (err) {
             console.log("error loading TODO items:", err);
         }
 
         console.log("existing todo items:", todoItems);
     });
-});
\ No newline at end of file
+
+    case 'delete':
```
Comment (zhaobn) [inline Week2/todo_app/program.js]: Where does your cases come from?

File: Week2/todo_app/program.js [hunk 5]
```diff
@@ -62,11 +123,121 @@ dbConnection.connect(function(err) {
...
+            break;
+        }
+
+    case 'tag':
+        {
+            todoModel.tagTodoItem(arg1, arg2, function (err) {
+                if (err) {
+                    console.log("error tagging TODO item:", err);
+                }
+                console.log('TODO item:', arg1, 'has been tagged to', arg2)
```
Comment (zhaobn) [inline Week2/todo_app/program.js]: Did you try your loggings here? Does it look like what you have in mind? Remember that you get the queryResult as a json object and you need some extra work to fit it within a sentence (as how you plan to log here)

**Pull request closed without merging**